### PR TITLE
Feature: role endpoints

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

File: src/test/resources/features/Role.feature [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    When I request "/roles/[role_id]" with method "PUT" with json
+      | key  | value               |
+      | name | Super administrator |
+    Then I expect "400" as status code
+    And I expect response fields length is "5"
+    And I expect response field "message" is "Wrong field value."
+    And I expect response field "code" is "206"
+    And I expect response field "field" is "constraint"
+    And I expect response field "value" is "access_controls_name_type_key"
+    And I expect response field "cause" is "NOT_NULL"
```
Comment (justinehell) [inline src/test/resources/features/Role.feature:68]: missing end line

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/main/java/com/ditrit/letomodelizerapi/model/accesscontrol/AccessControlType.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * Enum representing the types of access controls in a system.
+ * This enum is used to categorize different aspects of access control,
+ * defining how they interact with users, groups, and roles.
+ *
+ * <p>Each access control type has specific linkage rules:</p>
+ * <ul>
+ *     <li>ROLE - Represents a role, which is exclusively linked to permissions.
+ *         Roles are typically used to group sets of permissions and can be assigned to users or groups to grant those
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/main/java/com/ditrit/letomodelizerapi/model/accesscontrol/AccessControlType.java:11]: if you mean at the end of sentence, I can see them

File: src/test/java/com/ditrit/letomodelizerapi/cucumber/StepDefinitions.java [hunk 3]
```diff
@@ -348,31 +382,14 @@ public void setResponseFieldToContext(String from, String to) {
...
-
-    @Given("I clean token {string}")
-    public void cleanToken(String name) {
-        this.request(String.format("/users/token/%s", name), "DELETE");
+        this.clean("roles", String.format("name=%s", name));
     }
 
     public void clean(String entity, String query) throws URISyntaxException, IOException, InterruptedException {
         this.request(String.format("/%s?%s", entity, query));
+        System.out.println();
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/test/java/com/ditrit/letomodelizerapi/cucumber/StepDefinitions.java:392]: done

File: src/test/java/com/ditrit/letomodelizerapi/cucumber/StepDefinitions.java [hunk 4]
```diff
@@ -70,15 +71,47 @@ public void checkServerTrusted(X509Certificate[] arg0, String arg1) throws Certi
     private JsonNode resources;
     private JsonNode responseObject;
 
-    @Given("I use no authentication")
-    public void setNoAuthentication() {
-        authenticate = false;
-    }
+    @Given("I init admin")
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/test/java/com/ditrit/letomodelizerapi/cucumber/StepDefinitions.java:74]: done

File: src/test/resources/features/Role.feature (hunk 1 above)
Comment (Zorin95670) [reply src/test/resources/features/Role.feature:68]: done

File: src/main/java/com/ditrit/letomodelizerapi/controller/RoleController.java [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+     * @param id      the ID of the role to retrieve
+     * @return a Response containing the AccessControlDTO of the requested role
+     */
+    @GET
+    @Path("/{id}")
+    public Response getRoleById(final @Context HttpServletRequest request,
+                                final @PathParam("id") @Valid @NotNull Long id) {
+        HttpSession session = request.getSession();
+        User user = userService.getFromSession(session);
+        userPermissionService.checkIsAdmin(user, null);
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/controller/RoleController.java:106]: in another pr, not the purpose of this request. Should be comment in previous pr when I work on the acl...

**Review by justinehell — APPROVED** [review]

**Pull request merged**